src/libs/import.c

```diff
@@ -736,7 +747,19 @@ static guint _import_set_file_list(const gchar *folder, const int folder_lgth,
       if(include_jpegs || (ext && g_ascii_strncasecmp(ext, ".jpg", sizeof(".jpg"))
                                && g_ascii_strncasecmp(ext, ".jpeg", sizeof(".jpeg"))))
       {
-        const gboolean already_imported = dt_images_already_imported(fullname);
+        /* check if image is already imported, using previously fetched filroll id */
+        gboolean already_imported = FALSE;
+        if(filmroll_id != 0)
```

Comment:
Well that's mostly what `dt_images_already_imported()` does and we share the code. So better remove this code from here and use `dt_images_already_imported()`. The speed-up was from the folder check above.

Reply:
I just see we have post review at the same time :)
then this need `dt_images_already_imported()` to be modified in order to accept film id as parameter (that's what I suggest in my review...